peerconnection.go

```diff
@@ -1064,6 +1064,20 @@ func (pc *PeerConnection) SetRemoteDescription(desc SessionDescription) error {
 				pc.mu.Lock()
 				pc.addRTPTransceiver(t)
 				pc.mu.Unlock()
+
+				// if transceiver is create by remote sdp, set prefer codec same as remote peer
+				if codecs, err := codecsFromMediaDescription(media); err == nil {
+					filteredCodecs := []RTPCodecParameters{}
+					for _, codec := range codecs {
+						if c, matchType := codecParametersFuzzySearch(codec, pc.api.mediaEngine.getCodecsByKind(kind)); matchType == codecMatchExact {
+							// if codec match exact, use payloadtype register to mediaengine
+							codec.PayloadType = c.PayloadType
+							filteredCodecs = append(filteredCodecs, codec)
+						}
+					}
+					_ = t.SetCodecPreferences(filteredCodecs)
+				}
```

Comment:
The method `SetRemoteDescription` is already too big. What do you think about extracting added code to a function?
You can also remove comment and use method name `SetPreferCodecSameAsFromRemotePeerIfTransceiverIsCreateByRemoteSDP`.

Reply:
A function only for here and can't be reused in other place, I think it may not a better way..

Reply:
Correct. Why do you want it to be reusable? One can use functions to divide bigger functions to smaller parts.